EIPS/eip-6384.md

```diff
@@ -0,0 +1,155 @@
+
+The strings SHOULD NOT be formatted (e.g. should not contain HTML code) and wallets SHOULD treat this string as an untrusted input and handle its rendering as such.
+
+As it might be useful to provide a more structured output, we want to get the feedback of the community and specifically wallets' implementors on how they would like to receive it. One suggestion is that function would return structs that represent transfers and approvals to support a simulation like experience to the users. e.g. the `evalEIP712Buffer` function would return a strcut that represnts a transfer (contract, from, to, amount) and the wallet MAY display a resolved version of it to the end user. ("transfer 2 USDC to address 0x...")
+
+### Support for EIP-712 messages that are not meant to be used on-chain
+
+In case the EIP-712 message is not bounded to a contract, the wallet cannot call this function as the contract address is unkown and SHOULD default to its current display.
+
+Some web2 implementations are using a signed EIP-712 message as a method to identify the user. It would be helpful if the wallet can display a reassuring message that this message cannot be used on-chain.
+We would like to get feedback from the community if this can be done safely, e.g. by omitting the `chainId` and/or `verifyingContract` from EIP-712 message, or putting non-exisiting values in them (e.g. `verifyingContract` = 0xCcCCccccCCCCcCCCCCCcCcCccCcCCCcCcccccccC)
+
+## Rationale
+
+The proposed solution solves the readability issues. The incentives for keeping the EIP-712 message descritption as accurate as possible are alligned, as the responsibility for the description is now owned by the contract, that:
+
+- knows the message meaning exactly (and probably can reuse the code that handles this message when received on chain)
+- Natively incentivized to provide the best explanation to prevent a possible fraud
+- Not involving a third party that needs to be trusted
+- Maintains the fee-less customer experience as the added function is in “view” mode and does not require an on-chain execution and fees.
+- Maintains Web3’s composability property
+
+### Alternative solutions
+
+#### Third party services:
+
+Currently, the best choice for users is to rely on some 3rd party solutions that get the proposed message as input and explain its intended meaning to the user. This approach is:
+
+- Not scalable: 3rd party provider needs to learn all such proprietary messages
+- Not necessarily correct: the explanation is based on 3rd party interpretation of the original message author
+- Introduces an unnecessary dependency of a third party which may have some operational, security, and privacy implications.
+
+#### Domain name binding
+
+Alternatively, wallets can bind domain name to a signature. i.e. only accept EIP-712 message if it comes from a web2 domain that its `name` as defined by EIP-712 is included in `eip712Domain`. However this approach has the following disadvantages:
```

Comment:
I think there's a typo here. Should `that` be `than`?

Reply:
should be "that", adding parentheses to make it more clear